theano/gpuarray/opt.py

```diff
@@ -881,6 +881,11 @@ def local_gpua_softmaxwithbias(node, context_name):
 @register_opt('fast_compile')
 @op_lifter([theano.tensor.opt.Assert])
 def local_assert(node, context_name):
+
+    # Check if input nodes are already on the GPU
+    for i in node.inputs:
+        if isinstance(i.type, GpuArrayType):
+            return
```

Comment:
Here we only move the 1st input to the GPU, so the code don't need a for loop:

if isinstance(node.inputs[0].type, GpuArrayType):
   return